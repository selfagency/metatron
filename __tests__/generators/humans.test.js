const generateHumansTxt = require('../../dist/generators/humans').default
const { readFileSync } = require('atomically')

const humans = {
  site: {
    name: 'The Self Agency',
    languages: ['en', 'yi']
  },
  publisher: {
    name: 'The Self Agency',
    contact: 'hello@self.agency',
    url: 'https://self.agency',
    social_media: ['https://www.facebook.com/selfagency', 'https://www.instagram.com/selfagency']
  },
  contributors: [
    {
      name: 'Daniel Sieradski',
      role: 'Admin',
      contact: 'hello@self.agency',
      url: 'https://sieradski.co',
      social_media: ['https://www.facebook.com/selfagency', 'https://www.instagram.com/selfagency']
    },
    {
      name: 'Jane Doe',
      role: 'Editor',
      contact: 'hello@self.agency',
      url: 'https://example.com'
    }
  ],
  credits: [
    {
      name: 'Ribono Shel Olam',
      role: 'Master of the Universe',
      contact: 'https://shemayim.org/contact',
      url: 'https://shemayim.org'
    },
    {
      name: 'Shel Silverstein',
      role: 'Master of the Page',
      url: 'https://shelsilverstein.com'
    }
  ],
  stack: {
    components: ['typescript', 'nunjucks'],
    standards: ['n/a'],
    devtools: ['vscode']
  }
}

test('generate humans.txt - static', () => {
  const humansTxt = generateHumansTxt({
    ...humans,
    settings: {
      mode: 'generator',
      output_dir: './__tests__/__output__',
      tagline: true
    }
  })

  const output = readFileSync(`${process.cwd()}/__tests__/__output__/humans.txt`).toString()
  expect(output).toEqual(humansTxt)
})

test('generate humans.txt - middleware', () => {
  const humansTxt = generateHumansTxt({
    ...humans,
    settings: {
      mode: 'middleware',
      tagline: false
    }
  })

  expect(humansTxt).toMatchSnapshot()
})

test('generate humans.txt - missing', () => {
  expect(() =>
    generateHumansTxt({
      settings: {
        mode: 'middleware'
      }
    })
  ).toThrow()
})

test('generate humans.txt - invalid', () => {
  expect(() =>
    generateHumansTxt({
      settings: {
        mode: 'middleware'
      },
      humans: {
        publisher: {
          name: 'John Doe'
        }
      }
    }).toThrow()
  )
})
