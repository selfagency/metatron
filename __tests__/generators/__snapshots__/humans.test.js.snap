// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate humans.txt - middleware 1`] = `
"/* PUBLISHER */
The Self Agency
Contact: hello@self.agency
Webpage: https://self.agency

: 
: 



/* TEAM */
Admin: Daniel Sieradski
Contact: hello@self.agency
Webpage: https://sieradski.co

: 
: 



Editor: Jane Doe
Contact: hello@self.agency
Webpage: https://example.com

/* THANKS */
Master of the Universe: Ribono Shel Olam
Contact: https://shemayim.org/contact
Webpage: https://shemayim.org

Master of the Page: Shel Silverstein
Webpage: https://shelsilverstein.com

/* SITE */
Languages: en, yi
Components: typescript, nunjucks
Standards: n/a
Devtools: vscode

# Last updated date goes here
# humans.txt (https://humanstxt.org/)
# Made with Metatron (https://github.com/selfagency/metatron)

"
`;
